ineffassign.go

```diff
@@ -368,14 +368,17 @@ func isZeroInitializer(x ast.Expr) bool {
 		x = c.Args[0]
 	}
 
-	b, ok := x.(*ast.BasicLit)
-	if !ok {
-		return false
-	}
-	switch b.Value {
-	case "0", "0.0", "0.", ".0", `""`:
-		return true
+	switch x.(type) {
```

Comment:
Write this as `switch x := x.(type) {` and then you won't have to repeat the type assertions below.

Reply:
TIL! I didn't know you could do that. Looks much better.

Is this what you had in mind?

Reply:
Precisely 😄